crates/core/src/test_scenario.rs

```diff
@@ -505,6 +505,35 @@ where
             .wallet(&wallet)
             .network::<AnyNetwork>()
             .connect_http(rpc_url.to_owned());
+        let read_client = DynProvider::new(
```

Comment:
we don't actually need to instantiate a new provider here -- why not just use the existing `wallet_client`?